src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java

```diff
@@ -129,7 +129,8 @@ public void setUp() throws Exception {
      * @noinspection JUnitTestMethodWithNoAssertions, TestMethodWithoutAssertion
      * @noinspectionreason JUnitTestMethodWithNoAssertions - asserts in callstack,
      *      but not in this method
-     * @noinspectionreason TestMethodWithoutAssertion - until issue #14625
+     * @noinspectionreason TestMethodWithoutAssertion - asserts in callstack,
+     *      but not in this method
```

Comment:
We already have similar rule in some other tools, can you recheck if we suppressed this already for other tools maybe pmd or ....

Reply:
@romani , pmd yes....

https://github.com/checkstyle/checkstyle/blob/bb56e9de12e0f42c0188eb2a6cad839404d408c1/config/pmd-test.xml#L107-L110

Reply:
Can it help if we rename `examineCheckSection` to `assertCheckSection` ? May be we can extend config to register this method name as asset.

Reply:
> Can it help if we rename `examineCheckSection` to `assertCheckSection` ? May be we can extend config to register this method name as asset.

@romani , just pointing out, we have our methods named in chains.
 `examineCheckSection` is followed by `examineCheckSectionChildren` which in turn is followed by `examineCheckSubSection`

https://github.com/checkstyle/checkstyle/blob/1fc59870b6965480e2b1923ae6f9207fd71a7a30/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java#L165

https://github.com/checkstyle/checkstyle/blob/1fc59870b6965480e2b1923ae6f9207fd71a7a30/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java#L190

https://github.com/checkstyle/checkstyle/blob/1fc59870b6965480e2b1923ae6f9207fd71a7a30/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java#L206

Do we really want to extend our config to new method and hack all these just to workaround for one inspection that won't even impact users as much?

Please do correct me if I'm wrong, I don't mean to cross your suggestion 😃

Reply:
@romani , ping please.

Reply:
examineCheckSection is actually do assert by execution of
checker.process(files);

All Junit asserts are Custom Check  that helps to parse java files.
We can add assert for checker to have no violations or errors after execution to testAllCheckSectionJavaDocs() as it is a field of class, so no suppression will be required.